e2e_test/ddl/show_purify.slt

```diff
@@ -0,0 +1,20 @@
+# Test definition purification for `CREATE TABLE AS`, mainly focusing on the data types.
+statement ok
+create table ctas as select
+    0::int as v0,
+    1::decimal as v1,
+    '2022-03-13 01:00:00'::timestamp as v2,
+    '2022-03-13 01:00:00Z'::timestamptz as v3,
+    array['foo', 'bar', 'null'] as v4,
+    (1, (2, 3))::STRUCT<i BIGINT, j STRUCT<a BIGINT, b VARCHAR>> as v5,
+    hex_to_int256('0x11') as v6,
+    map{'key1': 1, 'key2': 2, 'key3': 3} as v7
+;
+
+query TT
+show create table ctas;
+----
+public.ctas	CREATE TABLE ctas (v0 INT, v1 NUMERIC, v2 TIMESTAMP, v3 TIMESTAMP WITH TIME ZONE, v4 CHARACTER VARYING[], v5 STRUCT<i BIGINT, j STRUCT<a BIGINT, b CHARACTER VARYING>>, v6 rw_int256, v7 MAP(CHARACTER VARYING,INT))
```

Comment:
After the change, how/is it possible (is it meaningful?) to get the "original create sql" for CTAS and schema registry? It seems useful for debugging/development purpose.

Reply:
It can be done via inspecting `definition` field of the metadata (like through the dashboard). However, I suppose it's indeed not that meaningful, as altering a CTAS will cause the purified definition to be persisted.

Reply:
> as altering a CTAS will cause the purified definition to be persisted.

Note: This is not the current behavior as purified definition is only used for `SHOW` but not the base for schema change replanning.